### PR TITLE
Improve json parsing for findStatus

### PR DESCRIPTION
**Pull request by mmatoszko**
Related to #378 

## Summary
This might be just a tip of the iceberg, but it seems like a good start.

There's one thing which is super unclear for me right now.

```swift
if firstStatus.caseSensitive != nil {
    query?.ignoreCase = !(statusQuery.caseSensitive != nil)
}
```

We first check the first item of the array if it has a value for `"case_sensitive"`, and then we get an other value under the same key from the actual index of the array.
Do we want to check the `status.first` for the presence of the value?

## Review Checklist
- [ ] I have responded to reviews and made changes where appropriate.
- [x] I have tested the code
- [ ] I have updated comments / documentation related to the changes I made.
- [ ] I have rebased my PR branch onto xi-mac/master.

### CODE REVIEW

**Comment by cmyr** [comment]
Sort of a higher level comment, but my thinking here is that we should actually have a `FindQuery` struct and a `fromJson` failable initializer, and we should parse these into a real type as they come in, doing validation at that point.

**Comment by mmatoszko** [comment]
Ok, I created `FindStatus` struct and changed the `findStatus(status: [[String: AnyObject]])` accordingly. I decided to not use `FindQuery`, because it's also used for `multi_find`, and the protocol definitions differ.

Feel free to take a look 😄

The question in [Summary](https://github.com/xi-editor/xi-mac/pull/379#issue-238937061) remains the same @scholtzan what do you think?

File: Sources/XiEditor/Search.swift [hunk 1]
```diff
@@ -388,59 +388,37 @@ extension EditViewController {
 
     func findStatus(status: [[String: AnyObject]]) {
         var findMarker: [Marker] = []
+        let statusStructs = status.flatMap(FindStatus.init)
```
Comment (cmyr) [inline Sources/XiEditor/Search.swift]: I think it would be good to know, somehow, if there is malformed data in here; that would indicate a bug in core.

Comment (mmatoszko) [reply Sources/XiEditor/Search.swift]: Do you mean an _integration_ test? 😄

Comment (cmyr) [reply Sources/XiEditor/Search.swift]: now that's an idea 🤔

**Pull request merged**